pkg/nullio/ls2json.go

```diff
@@ -163,11 +163,14 @@ func (w *Ls2JsonWriter) parseFileHeader(matches [][]byte) ([]byte, error) {
 	if entryType == dto.EntryTypeLink {
 		parts := strings.Split(string(name), " -> ")
 		const NumberOfPartsInALink = 2
-		if len(parts) == NumberOfPartsInALink {
+		switch len(parts) {
+		case NumberOfPartsInALink:
 			name = dto.FilePath(parts[0])
 			linkTarget = dto.FilePath(parts[1])
-		} else {
-			log.WithContext(w.sentrySpan.Context()).Error("could not split link into name and target")
+		case 1:
+			// This case happens when a user tries to read the target of a link without permission. See #596.
```

Comment:
We could improve the comment to notice that we don't do any action for such an entry (no op).